apps/extension/src/contents/inpage.ts

```diff
@@ -40,21 +40,23 @@ export const windowAI: WindowAI<ModelID> = {
   async getCompletion(input, options = {}) {
     const { onStreamResult } = _validateOptions(options)
     const shouldStream = !!onStreamResult
-    const shouldReturnMultiple = options.numOutputs && options.numOutputs > 1
     const requestId = _relayRequest(PortName.Completion, {
       transaction: transactionManager.init(input, _getOriginData(), options),
       shouldStream
     })
     return new Promise((resolve, reject) => {
-      _addResponseListener<CompletionResponse>(requestId, (res) => {
-        if (isOk(res)) {
-          resolve(shouldReturnMultiple ? res.data : res.data[0])
-          onStreamResult && onStreamResult(res.data[0], null)
-        } else {
-          reject(res.error)
-          onStreamResult && onStreamResult(null, res.error)
+      _addResponseListener<CompletionResponse<typeof input>>(
+        requestId,
+        (res) => {
+          if (isOk(res)) {
+            resolve(res.data)
```

Comment:
since this is going to break all the apps I believe, can we leave the extension as-is and just modify the ts lib for now to have it always return an array?

Reply:
This one is rough - I'm trying to figure out what's the best way to make the type infer work, but couldn't figure out how to get it working for the return type... will need to use any to escape some of the type.